pkg/controller/remotemachineset/awsactuator.go

```diff
@@ -216,24 +253,90 @@ func (a *AWSActuator) updateProviderConfig(machineSet *machineapi.MachineSet, in
 }
 
 // getSubnetsByAvailabilityZones maps availability zones to subnet
-func (a *AWSActuator) getSubnetsByAvailabilityZone(subnets []string) (map[string]string, error) {
+func (a *AWSActuator) getSubnetsByAvailabilityZone(pool *hivev1.MachinePool) (map[string]string, error) {
 	subnetsByAvailabilityZone := map[string]string{}
-	idPointers := make([]*string, len(subnets))
-	for i, id := range subnets {
+	idPointers := make([]*string, len(pool.Spec.Platform.AWS.Subnets))
+	for i, id := range pool.Spec.Platform.AWS.Subnets {
 		idPointers[i] = aws.String(id)
 	}
 
-	results, err := a.client.DescribeSubnets(&ec2.DescribeSubnetsInput{SubnetIds: idPointers})
+	results, err := a.awsClient.DescribeSubnets(&ec2.DescribeSubnetsInput{SubnetIds: idPointers})
 	if err != nil {
+		conds, changed := controllerutils.SetMachinePoolConditionWithChangeCheck(
+			pool.Status.Conditions,
+			hivev1.InvalidSubnetsMachinePoolCondition,
+			corev1.ConditionTrue,
+			"SubnetsNotFound",
+			err.Error(),
```

Comment:
Sample message for this condition -> `InvalidSubnetID.NotFound: The subnet ID 'subnet-1' does not exist status code: 400, request id: f506c630-ab03-4d81-b116-745e0b3e36ce`. Do we need to format it?

Reply:
We have to do some formatting. If we leave it as is, then the condition will change every time since the request id will change.

Also, what do you think about only setting the condition when the error code is "InvalidSubnetID.NotFound"?